File: backend/schemas/import_export.py
"""Schemas related to import/export and backup workflows."""

from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class ExportConfig(BaseModel):
    """Export configuration schema shared across services."""

    loras: bool = False
    lora_files: bool = False
    lora_metadata: bool = False
    lora_embeddings: bool = False
    generations: bool = False
    generation_range: str = "all"
    date_from: Optional[str] = None
    date_to: Optional[str] = None
    user_data: bool = False
    system_config: bool = False
    analytics: bool = False
    format: str = "zip"
    compression: str = "balanced"
    split_archives: bool = False
    max_size_mb: int = 1024
    encrypt: bool = False
    password: Optional[str] = None


class ExportEstimate(BaseModel):
    """Export size and time estimates."""

    size: str
    time: str


class ImportConfig(BaseModel):
    """Import configuration schema."""

    mode: str = "merge"
    conflict_resolution: str = "ask"
    validate: bool = True
    backup_before: bool = True
    password: Optional[str] = None


class BackupHistoryItem(BaseModel):
    """Backup history item schema."""

    id: str
    created_at: datetime
    type: str
    size: Optional[int] = Field(default=None, ge=0)
    status: str


class BackupCreateRequest(BaseModel):
    """Request body for initiating a backup operation."""

    backup_type: str = Field(default="full", alias="backup_type")

    model_config = {
        "populate_by_name": True,
    }

File: app/frontend/src/types/importExport.ts
/**
 * Types derived from backend/schemas/import_export.py for backup, export and import flows.
 */

import type { BackendSchemas } from './generated';

type Schemas = BackendSchemas;

export type ExportGenerationRange = Schemas['ExportConfig']['generation_range'];

export type ExportFormat = Schemas['ExportConfig']['format'];

export type ExportCompression = Schemas['ExportConfig']['compression'];

export type ExportConfig = Schemas['ExportConfig'];

export type ExportEstimate = Schemas['ExportEstimate'];

export type ImportMode = 'merge' | 'replace' | 'skip';

export type ImportConflictResolution = 'ask' | 'keep_existing' | 'overwrite' | 'rename';

export interface ImportConfig {
  mode: ImportMode;
  conflict_resolution: ImportConflictResolution;
  validate: boolean;
  backup_before: boolean;
  password: string;
}

export type BackupHistoryItem = Schemas['BackupHistoryItem'];

export type BackupCreateRequest = Schemas['BackupCreateRequest'];
